### PR TITLE
#12Logback_xml

### PR DESCRIPTION
**Pull request by AbseliamovEnver**
Update logback.xml

### CODE REVIEW

File: src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+
+    </appender>
+
+    <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+
+    </appender>
+
+    <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <layout class="ch.qos.logback.classic.PatternLayout">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </layout>
+    </appender>
+
+    <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/resources/logback.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+
+    </appender>
+    <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+
+    </appender>
+    <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 7]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        <layout class="ch.qos.logback.classic.PatternLayout">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </layout>
+    </appender>
+    <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+    </appender>
+
+    <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 10]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+    </appender>
+
+    <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 12]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <layout class="ch.qos.logback.classic.PatternLayout">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </layout>
+    </appender>
+
+    <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+            <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/debug.%d{yyyy-MM-dd}%i.log.zip</fileNamePattern>
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+    </appender>
+    <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/error.%d{yyyy-MM-dd}%i.log.zip</fileNamePattern>
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+    </appender>
+    <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 15]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        <layout class="ch.qos.logback.classic.PatternLayout">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </layout>
+    </appender>
+    <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+                    <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/debug.%d{yyyy-MM-dd}%i.log.zip</fileNamePattern>
+                    <minIndex>1</minIndex>
+                    <maxIndex>10</maxIndex>
+                </rollingPolicy>
+
+                <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+                    <maxFileSize>20MB</maxFileSize>
+                </triggeringPolicy>
+            </appender>
+            <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+                    <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/error.%d{yyyy-MM-dd}%i.log.zip</fileNamePattern>
+                    <minIndex>1</minIndex>
+                    <maxIndex>10</maxIndex>
+                </rollingPolicy>
+
+                <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+                    <maxFileSize>20MB</maxFileSize>
+                </triggeringPolicy>
+            </appender>
+            <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 19]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+            <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+                <layout class="ch.qos.logback.classic.PatternLayout">
+                    <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+                </layout>
+            </appender>
+            <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+            <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/debug.%d{yyyy-MM-dd}%i.log.zip</fileNamePattern>
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+   </appender>
+   <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 22]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+   <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+       <layout class="ch.qos.logback.classic.PatternLayout">
+           <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+       </layout>
+   </appender>
+   <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 24]
```diff
@@ -0,0 +1,69 @@
...
+        <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
+            <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/error.%d{yyyy-MM-dd}%i.log.zip</fileNamePattern>
+            <minIndex>1</minIndex>
+            <maxIndex>10</maxIndex>
+        </rollingPolicy>
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+   </appender>
+   <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: pom.xml [hunk 25]
```diff
@@ -27,6 +27,18 @@
             <artifactId>commons-lang3</artifactId>
             <version>3.5</version>
         </dependency>
+		
+		<dependency>
+            <groupId>org.slf4j</groupId>
```
Comment (zozich) [inline pom.xml]: Do you really need Slf4j here for Logback to work?

File: src/main/resources/logback.xml (hunk 22 above)
Comment (zozich) [inline src/main/resources/logback.xml]: You have indentation of 3 in this file. Make it 4

File: src/main/resources/logback.xml [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+   <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>JavaCoreFinalProjectGroup6/logs/application.log</file>
+        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </encoder>
+
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>JavaCoreFinalProjectGroup6/logs/archived/debug.%d.log</fileNamePattern>
+            <timeBasedFileNamingAndTriggeringPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP">
+                <maxHistory>30</maxHistory>
```
Comment (zozich) [inline src/main/resources/logback.xml]: Remove all configs that are not directly required by the specification

File: src/main/resources/logback.xml [hunk 27]
```diff
@@ -0,0 +1,69 @@
...
+                <maxFileSize>50MB</maxFileSize>
+            </timeBasedFileNamingAndTriggeringPolicy>
+        </rollingPolicy>
+
+        <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
```
Comment (zozich) [inline src/main/resources/logback.xml]: Do you really need several rolling policies here?

File: src/main/resources/logback.xml [hunk 28]
```diff
@@ -0,0 +1,69 @@
...
+        </rollingPolicy>
+
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+   </appender>
+   <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
+            <level>ERROR</level>
+            <onMatch>ACCEPT</onMatch>
```
Comment (zozich) [inline src/main/resources/logback.xml]: Is it really needed? Isn't just level enough?

**Review by zozich — CHANGES_REQUESTED** [review]
Please amend

File: src/main/resources/logback.xml [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+            <fileNamePattern>logs/archivelogs/error.%i.log</fileNamePattern>
+        </rollingPolicy>
+        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+            <maxFileSize>20MB</maxFileSize>
+        </triggeringPolicy>
+        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </encoder>
+    </appender>
+    <root level="INFO">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml:33]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+    <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>logs/application.log</file>
+        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </encoder>
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>logs/archivelogs/info.%d.gz</fileNamePattern>
+        </rollingPolicy>
+    </appender>
+    <appender name="FILE-ERROR" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml:18]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 31]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <statusListener class="ch.qos.logback.core.status.NopStatusListener"/>
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml:3]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml:4]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

File: src/main/resources/logback.xml [hunk 33]
```diff
@@ -0,0 +1,38 @@
...
+        <layout class="ch.qos.logback.classic.PatternLayout">
+            <Pattern>%d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</Pattern>
+        </layout>
+    </appender>
+    <appender name="FILE-AUDIT" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Sheptytskyid) [inline src/main/resources/logback.xml:9]: MINOR Make this line start at column 3. [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIndentCheck)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by zozich — APPROVED** [review]
Cool!

**Pull request merged**